packages/typespec-azure-resource-manager/test/rules/no-response-body.test.ts

```diff
@@ -33,10 +35,46 @@ describe("typespec-azure-resource-manager: no response body rule", () => {
       .toBeValid();
   });
 
+  it("is valid if 204 responses have no body", async () => {
+    await tester
+      .expect(
+        `
+        namespace Azure.MyService;
+
+        model TestAcceptedResponse {
+          @statusCode statusCode: 204;
+        }
+        op walk(): TestAcceptedResponse;
+      `,
+      )
+      .toBeValid();
+  });
+
+  it("emit warnings if a 204 response has a body", async () => {
+    await tester
+      .expect(
+        `
+        namespace Azure.MyService;
+        
+        model TestAcceptedResponse {
```

Comment:
I would use the arm response templates in these tests

Reply:
Not sure how this is emitting diagnostices if we are skipping namespaces starting with `Azure`

Reply:
we are skipping namespaces that doesn't start with `Azure`